src/deluge/model/instrument/non_audio_instrument.cpp

```diff
@@ -85,8 +100,13 @@ void NonAudioInstrument::sendNote(ModelStackWithThreeMainThings* modelStack, boo
 		// Run everything by the Arp...
 		arpeggiator.noteOn(arpSettings, noteCodePreArp, velocity, &instruction, fromMIDIChannel, mpeValues);
 
-		if (instruction.noteCodeOnPostArp != ARP_NOTE_NONE) {
-			noteOnPostArp(instruction.noteCodeOnPostArp, instruction.arpNoteOn);
+		for (int32_t n = 0; n < ARP_MAX_INSTRUCTION_NOTES; n++) {
+			if (instruction.arpNoteOn != nullptr && instruction.arpNoteOn->noteCodeOnPostArp[n] != ARP_NOTE_NONE) {
```

Comment:
```suggestion
		if (instruction.arpNoteOn != nullptr) {
			for (int32_t n = 0; n < ARP_MAX_INSTRUCTION_NOTES; n++) {
				if (instruction.arpNoteOn->noteCodeOnPostArp[n] == ARP_NOTE_NONE) { break; }
```

The optimizer is probably already doing this transform, but IMO this improves readability.